src/Aspire.Dashboard/Components/Pages/TraceDetail.razor.cs

```diff
@@ -197,7 +197,7 @@ private void UpdateDetailViewData()
         }
 
         Logger.LogInformation("Trace '{TraceId}' has {SpanCount} spans.", _trace.TraceId, _trace.Spans.Count);
-        _spanWaterfallViewModels = SpanWaterfallViewModel.Create(_trace, new SpanWaterfallViewModel.TraceDetailState(_collapsedSpanIds));
+        _spanWaterfallViewModels = SpanWaterfallViewModel.Create(_trace, new SpanWaterfallViewModel.TraceDetailState(OutgoingPeerResolvers.ToArray(), _collapsedSpanIds));
```

Comment:
Why did you choose an array as the parameter vs an enumerable?

Reply:
To avoid allocating a new enumerator each time it is iterated over.